### ISSUE TITLE
Investigate known performance issues with gRPC-Netty and consider alternatives

### ISSUE THREAD

**Issue by adammitchelldev** [vaticle/typedb#5769]
## Description

GRPC-Netty is known to have some performance issues which can lead to ~5x slower streaming performance:

https://github.com/grpc/grpc-java/issues/6696

We should investigate the performance difference between gRPC-Netty and other implementation such as as gRPC-HttpOk.

## Further Information

In upgrading gRPC from 1.24 to 1.29, we experienced a very notable ~5x performance drop on various benchmarks (including simulation) and have had to revert the upgrade. Seeing as this is within the same order of magnitude, we might suspect that changes to gRPC have caused this performance issue to suddenly surface.

**Comment by lriuui0x0** [comment]
We have tested this problem against 2.0 with different versions of grpc and netty, without noticeable differences. We did the same tests against 1.8, and the 5x slower problem is indeed there. Therefore we suspect the problem arises only from the specific way we use grpc-netty in 1.8 and the problem is not there anymore.

The code we used to test against 2.0:

```
        LOG.info("defining schema");
        localhostGraknTx(tx -> {
            final GraqlDefine defineQuery = Graql.define(
                type("person").sub("entity")
            );
            tx.query().define(defineQuery);
            tx.commit();
        }, Session.Type.SCHEMA);
        LOG.info("finished defining schema");

        int dataSize = 100000;

        LOG.info("inserting data");
        localhostGraknTx(tx -> {
            for (int i = 0; i < dataSize; i++) {
                final GraqlInsert insertQuery = Graql.insert(
                        var().isa("person")
                );
                tx.query().insert(insertQuery);
            }
            tx.commit();
        }, Session.Type.DATA);
        LOG.info("finished inserting data");

        LOG.info("reading data");
        localhostGraknTx(tx -> {
            Stream<? extends Entity> persons = tx.concepts().getEntityType("person").asRemote(tx).getInstances();
            persons.forEach(person -> {
                LOG.info("reading person");
            });
        }, Session.Type.DATA);
        LOG.info("finished reading data");
```

The code we used to test against 1.8:

```
        long currTime = System.currentTimeMillis();
        GraknClient client = new GraknClient("localhost:48555");
        GraknClient.Session session = client.session("grakn");

        GraknClient.Transaction tx = session.transaction().write();
        int dataSize = 65535;
        for (int i = 0; i < dataSize; i++) {
            tx.execute(Graql.insert(Graql.var().isa("person")));
        }
        tx.commit();

        tx = session.transaction().read();
        Stream<Entity.Remote> persons = tx.getEntityType("person").asRemote(tx).instances();
        persons.forEach(person -> System.out.println("hello"));

        long duration = System.currentTimeMillis() - currTime;
        System.out.println(duration);
```

The commit versions we used to test 1.8:

slow version (one commit before downgrade netty): https://github.com/graknlabs/grakn/commit/c4a444c06e948e436c60fda8626a933da94fcefe

fast version (1.8.4): https://github.com/graknlabs/grakn/commit/cf4de7f936bec2a06e63e4dc3d74543a40e86e99

**Comment by flyingsilverfin** [comment]
Closed with #5831
